src/main/webapp/src/app/nav-bar/nav-bar.component.spec.ts

```diff
@@ -0,0 +1,67 @@
+import { ComponentFixture, TestBed, inject, async } from '@angular/core/testing';
+import { NavBarComponent } from './nav-bar.component';
+import { Location } from '@angular/common';
+import { By } from '@angular/platform-browser';
+import { CommonModule } from '@angular/common';
+import { RouterTestingModule } from '@angular/router/testing';
+import { routes } from '../app-routing.module'
+
+describe('NavBarComponent', () => {
+ 
+  let component: NavBarComponent;
+  let fixture: ComponentFixture<NavBarComponent>;
+  let location: Location;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [ RouterTestingModule ],
```

Comment:
Would it work if you combine the both TestBed.configureTestingModule together? Because you both imports RouterTestingModule twice, does it work if you just import the one with 'withRoutes'? 
Link: https://angular.io/api/router/testing/RouterTestingModule

Reply:
fixed